dwave/system/samplers/dwave_sampler.py

```diff
@@ -562,9 +562,17 @@ def _hook(computation):
         if hasattr(computation, 'id'):
             info.update(problem_id=computation.id)
 
-        return dimod.SampleSet.from_samples((samples, variables), info=info, vartype=vartype,
-                                            energy=result['energies'],
-                                            num_occurrences=result.get('num_occurrences', None),
-                                            sort_labels=True)
+        sampleset = dimod.SampleSet.from_samples((samples, variables), info=info, vartype=vartype,
+                                                 energy=result['energies'],
+                                                 num_occurrences=result.get('num_occurrences', None),
+                                                 sort_labels=True)
+
+        if warninghandler is not None:
+            warninghandler.too_few_samples(sampleset)
+
+            if warninghandler.action is WarningAction.SAVE:
+                sampleset.info['warnings'] = warninghandler.saved
```

Comment:
the is overwriting the warnings already saved https://github.com/dwavesystems/dwave-system/pull/256/files#diff-589838e92ba22167debae5a20bf4f7a6R371. Probably we should remove that line. That also means that we don't need to pass in the info to `_result_to_response_hook`